src/sentry/issues/endpoints/organization_group_search_views.py

```diff
@@ -132,31 +142,70 @@ def get(self, request: Request, organization: Organization) -> Response:
-        if visibility:
-            org_query = GroupSearchView.objects.filter(
-                organization=organization,
-                visibility=GroupSearchViewVisibility.ORGANIZATION,
-            ).prefetch_related("projects")
-
-            owner_query = GroupSearchView.objects.filter(
+        if serializer.validated_data.get("visibility"):
+            visibility = serializer.validated_data.get(
+                "visibility", [GroupSearchViewVisibility.OWNER]
+            )
+            sort = serializer.validated_data.get("sort", "-last_seen")
+            user_starred_views = GroupSearchViewStarred.objects.filter(
                 organization=organization,
                 user_id=request.user.id,
-                visibility=GroupSearchViewVisibility.OWNER,
-            ).prefetch_related("projects")
+            ).values_list("group_search_view_id", flat=True)
+
+            owner_query_starred = (
+                GroupSearchView.objects.filter(
+                    organization=organization,
+                    user_id=request.user.id,
+                    visibility=GroupSearchViewVisibility.OWNER,
+                    id__in=user_starred_views,
+                )
+                .order_by(SORT_MAP[sort])
+                .prefetch_related("projects")
+            )
+
+            owner_query_non_starred = (
+                GroupSearchView.objects.filter(
+                    organization=organization,
+                    user_id=request.user.id,
+                    visibility=GroupSearchViewVisibility.OWNER,
+                )
+                .exclude(id__in=user_starred_views)
+                .order_by(SORT_MAP[sort])
+                .prefetch_related("projects")
+            )
+
+            org_query_starred = (
+                GroupSearchView.objects.filter(
+                    organization=organization,
+                    visibility=GroupSearchViewVisibility.ORGANIZATION,
+                    id__in=user_starred_views,
+                )
+                .order_by(SORT_MAP[sort])
+                .prefetch_related("projects")
+            )
+
+            org_query_non_starred = (
+                GroupSearchView.objects.filter(
+                    organization=organization,
+                    visibility=GroupSearchViewVisibility.ORGANIZATION,
+                )
+                .exclude(id__in=user_starred_views)
+                .order_by(SORT_MAP[sort])
+                .prefetch_related("projects")
+            )
 
-            param_query_map = {
-                GroupSearchViewVisibility.ORGANIZATION: org_query,
-                GroupSearchViewVisibility.OWNER: owner_query,
-            }
+            owner_queries = [owner_query_starred, owner_query_non_starred]
+            org_queries = [org_query_starred, org_query_non_starred]
 
-            query_list = [param_query_map[v] for v in visibility]
-            query = reduce(or_, query_list)
+            query_sources = [
```

Comment:
i would personally create a helper function called `get_query_sources` which would encapsulate the logic here & above. ([extract method](https://refactoring.guru/extract-method)).